src/api/app/models/flag.rb

```diff
@@ -70,6 +70,40 @@ def default_status
         all_flag || arch_flag
       all_flag = main_object.project.flags.where("flag = ? AND repo IS NULL AND architecture_id IS NULL", flag).first unless all_flag
     end
+    
+    if same_flag
+      return repo_flag.status if repo_flag
+      return arch_flag.status if arch_flag
+      return all_flag.status if all_flag
+    end
+    if arch_flag
+      return all_flag.status if all_flag
+    end
+    if repo_flag
+      return all_flag.status if all_flag
+    end
+    if main_object.kind_of? Package
+      return all_flag.status if all_flag
+    end
+    return Flag.default_status(flag)
+  end
+
+  def effective_status
+    all_flag = main_object.flags.where("flag = ? AND repo IS NULL AND architecture_id IS NULL", flag).first
+    repo_flag = main_object.flags.where("flag = ? AND repo = ? AND architecture_id IS NULL", flag, repo).first
+    arch_flag = main_object.flags.where("flag = ? AND repo IS NULL AND architecture_id = ?", flag, architecture_id).first
+    same_flag = main_object.flags.where("flag = ? AND repo = ? AND architecture_id = ?", flag, repo, architecture_id).first
+    # Package settings only override project settings...
+    if main_object.kind_of? Package
+      # do the same_flag check first to see if all_flag or same_flag had been set on package level, they *both* overwrite the project level
+      same_flag = main_object.project.flags.where("flag = ? AND repo = ? AND architecture_id = ?", flag, repo, architecture_id).first unless
+        all_flag || same_flag || repo_flag || arch_flag
+      repo_flag = main_object.project.flags.where("flag = ? AND repo = ? AND architecture_id IS NULL", flag, repo).first unless
+        all_flag || repo_flag || arch_flag
+      arch_flag = main_object.project.flags.where("flag = ? AND repo IS NULL AND architecture_id = ?", flag, architecture_id).first unless
+        all_flag || arch_flag
+      all_flag = main_object.project.flags.where("flag = ? AND repo IS NULL AND architecture_id IS NULL", flag).first unless all_flag
+    end
```

Comment:
This part looks like a copy & paste from default_status. If so, please move that part into a method and call it in effective_status and default_status